### ISSUE TITLE
2.0 Release - Feedback wanted 

### ISSUE THREAD

**Issue by lumoe** [lumoe/obsidian-rollover-daily-todos#80]
In order to increase maintenance and future development of the plugin, it will undergo a complete rewrite. The improvements include (1) modularization of the codebase, (2) switch to TypeScript, (3) use pnpm as the main package manager and (4) add unit testing. @tim-hilt has already implemented most of the things in his fork and we will start the integration in the following procedure. 

1. @tim-hilt s work will be integrated in the [`release-2.0`](https://github.com/lumoe/obsidian-rollover-daily-todos/pull/79) branch 
2. New version will be developed in this branch 
3. We want to invite as many testers as possible to test out the new release and provide us feedback. 

**Reporting errors template**
1. Obsidian version:
2. **All** settings from your _Daily notes_ or _Periodic Notes_ plugin :
3. Commit Hash: 

**Installation instructions for new release:** 
(1) Download or clone the `release-2.0` branch into your `path/to/vault/.obsidian/plugins/` (2) run `pnpm install` and (3) `pnpm run dev` will build the project.

**Comment by josephdong1000** [comment]
Version 2.0 addresses heading category rollover well!

One feature I do miss though is the ability to rollover all miscellaneous todos under one heading. Currently, you specify which headings act as source and targets for todos. Having another field where you define where all remaining todos will go would be nice.

Thanks, and cheers!

**Comment by tim-hilt** [comment]
Yes, that’s true… previous behavior isn’t possible anymore. Do you have a suggestion about how this could be realized in the settings? I don’t have a good looking/intuitive solution in mind just yet

**Comment by josephdong1000** [comment]
I had in mind whatever the field was before, IIRC it was just a text box where you typed the heading or a dropdown menu with all of the headings in the daily note template file.

The latter doesn't sound too bad to me. I'm imaging a toggle switch called "Gather miscellaneous todos" that defaults on and shows a dropdown with all headings in the todo file. That way people can choose to dump misÎc todos to a heading that already sources and receives todos (what I do), dump them to a different section, or ignore misc todos altogether.

**Comment by Agent-E11** [comment]
@josephdong1000 another possible solution is a dropdown which selects what to do with todos under each heading:
```md
<!-- Todo headings -->
# Heading 1      -      ignore
# Heading 2      -      pass-through
# Heading 3      -      consolidate
# Heading 4      -      consolidate 

<!-- Consolidate to: -->
# Heading 5
```
- Todos under `Heading 1` are ignored
- Todos under `Heading 2` are rolled over to `Heading 2` in the new note
- Todos under `Heading 3` and `Heading 4` are consolidated to `Heading 5`

**Comment by mariomui** [comment]
The changes should be introduced incrementally start from the new build system. stability over features.

**Comment by marcellonovak** [comment]
I wish there was either a method to detect todos under a heading, or a way to give the user an option of a special character to use before/after a checkbox to denote rollover, so that some can get rolled over and others won't

**Comment by WishUponATardis** [comment]
I'm new to git, and obsidian but more familiar there, and I'm not sure what this branch etc stuff is... any chance of a simplified breakdown of how to update this in full for newest? Why wouldn't updates from the community plugin section have done these? :s

**Comment by bl0ggi** [comment]
Would be awesome if you could select a line of text instead of a heading to put rollovers under. I currently use callouts for tasks and would love to be able to rollover new tasks to a callout.
